test/compilable/issue21340.d

```diff
@@ -11,11 +15,11 @@ extern(C++) class Foo {}
 extern(C++) void test(ScopeClass!Foo)
 {
 }
-version(Posix)
+version(CppMangle_Itanium)
 {
     static assert (test.mangleof == "_Z4testP10ScopeClassIP3FooE");
 }
-else version (CppRuntimeMicrosoft)
```

Comment:
That was a bug, wasn't it ?

Reply:
Yes, missing the underscore.  We'll see whether or not testsuite breaks as a result. :-)

Reply:
Yes, it uncovered an issue in the test:
```
compilable\issue21340.d(24): Error: static assert:  `"?test@@YAXPAV?$ScopeClass@PAVFoo@@@@@Z" == "?test@@YAXPEAV?$ScopeClass@PEAVFoo@@@@@Z"` is false
```

Reply:
Blindly assuming that this failure only applied to Win32.